### DISCUSSION TITLE
[firebase_crashlytics] Enable Firebase Crashlytics at runtime

### DISCUSSION THREAD

**Discussion by lschmierer** [firebase/flutterfire#5581] [Feature request]
To be compliant with European privacy laws, we are required to get user consent before transmitting data to Crashlytics.
We want to use Crashlytics as described at https://firebase.google.com/docs/crashlytics/customize-crash-reports?platform=ios#enable_opt-in_reporting

This is, however, not possible with the Flutter library at the moment, as Fabric is always initialised as soon as the plugin is attached to the engine.

A possible fix would be to wait for the first MethodCall to be received by the plugin and initialiseFabric after that.
That way a library user can take care to not make any calls to Crashlytics before user consent is given.

**Comment by Jjagg** [comment]
This has been resolved in the Crashlytics update: https://firebase.flutter.dev/docs/crashlytics/usage#enable-opt-in-reporting

**Reply by martin-braun** [reply]
Please re-open the issue, it is not solved. Yes, you can now disable data collection by the framework, but my software firewall still detects a connection from my app to firebaselogging-pa.googleapis.com and firebase-settings.crashlytics.com prior interacting with the Firebase SDKs. 

I'm not interested in promises by the service provider (Google) that no data collection will happen. I want my users to only connect to my service until they are fully aware about Firebase and acknowledge it.

Thanks.
